Source/AviationWeather.Test/Source/metar_parser_tests.cpp

```diff
@@ -0,0 +1,575 @@
+
+void MetarParserTests::METAR_Parser_Visibility()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(distance_unit::feet, metar.visibility_group.unit);
+    Assert::AreEqual(static_cast<double>(UINT32_MAX), metar.visibility_group.distance);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.visibility_group.modifier);
+
+    // M1/4SM
+    std::string m1("M1/4SM ");
+    metar::parsers::parse_visibility(metar, m1);
+    Assert::AreEqual(distance_unit::statute_miles, metar.visibility_group.unit);
+    Assert::AreEqual(1.0 / 4.0, metar.visibility_group.distance, 0.00001);
+    Assert::AreEqual(metar::visibility_modifier_type::less_than, metar.visibility_group.modifier);
+
+    // 1/2SM
+    std::string m2("1/2SM ");
+    metar::parsers::parse_visibility(metar, m2);
+    Assert::AreEqual(distance_unit::statute_miles, metar.visibility_group.unit);
+    Assert::AreEqual(1.0 / 2.0, metar.visibility_group.distance, 0.00001);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.visibility_group.modifier);
+
+    // 1SM
+    std::string m3("1SM ");
+    metar::parsers::parse_visibility(metar, m3);
+    Assert::AreEqual(distance_unit::statute_miles, metar.visibility_group.unit);
+    Assert::AreEqual(1.0, metar.visibility_group.distance, 0.00001);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.visibility_group.modifier);
+
+    // 1 3/4SM
+    std::string m4("1 3/4SM ");
+    metar::parsers::parse_visibility(metar, m4);
+    Assert::AreEqual(distance_unit::statute_miles, metar.visibility_group.unit);
+    Assert::AreEqual(1.0 + (3.0 / 4.0), metar.visibility_group.distance, 0.00001);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.visibility_group.modifier);
+
+    // 5/16SM
+    std::string m5("5/16SM ");
+    metar::parsers::parse_visibility(metar, m5);
+    Assert::AreEqual(distance_unit::statute_miles, metar.visibility_group.unit);
+    Assert::AreEqual(5.0 / 16.0, metar.visibility_group.distance, 0.00001);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.visibility_group.modifier);
+
+    // 3SM
+    std::string m6("3SM ");
+    metar::parsers::parse_visibility(metar, m6);
+    Assert::AreEqual(distance_unit::statute_miles, metar.visibility_group.unit);
+    Assert::AreEqual(3.0, metar.visibility_group.distance, 0.00001);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.visibility_group.modifier);
+
+    // 15SM
+    std::string m7("15SM ");
+    metar::parsers::parse_visibility(metar, m7);
+    Assert::AreEqual(distance_unit::statute_miles, metar.visibility_group.unit);
+    Assert::AreEqual(15.0, metar.visibility_group.distance, 0.00001);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.visibility_group.modifier);
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_RunwayVisualRange()
```

Comment:
Do non-variable RVR groups work with visibility modifiers?  Here are AC 00-45G examples:

Notation | Meaning
-------- | -------
R01L/0800FT | Runway 01L of 800 feet
R01L/0600V1000FT | runway 01L varying between 600 and 1,000 feet
R01L/M0600FT | runway 01L of less than 600 feet
R27/P6000FT | runway 27 of greater than 6,000 feet